### ISSUE TITLE
Feature: another default smearing parameters?

### ISSUE THREAD

**Issue by WHUweiqingzhou** [deepmodeling/abacus-develop#3310]
### Background

Current default smearing:
```cpp    
    occupations = "smearing"; 
    smearing_method = "fixed";
    smearing_sigma = 0.01;
```
`fixed` is bad choice for metal, and users have to set smearing-related parameters manually, which is not user-friendly. I suggest we change default to a more general one:
```cpp    
    occupations = "smearing"; 
    smearing_method = "gaussian";
    smearing_sigma = 0.01;
```

### Describe the solution you'd like

change `smearing_method` default to `gaussian`.

### Task list only for developers

- [ ] Notice possible changes of behavior
- [ ] Explain the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi

### Task list for Issue attackers (only for developers)

- [ ] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.

**Comment by QuantumMisaka** [comment]
Unlike VASP, the `smearing_sigma` in ABACUS have Rydberg unit, so is the default `smearing_sigma` set to `0.001` better ?

**Comment by WHUweiqingzhou** [comment]
@QuantumMisaka,
Yes, ABACUS's energy unit is Rydberg. Indeed, 0.01 Ry is not a small value. I think this issue should be discussed widely. And a reasonable test report is helpful for this discussion.
b.t.w., according to VASP's documentation [SIGMA](https://www.vasp.at/wiki/index.php/SIGMA), whose default is a larger one 0.2 eV.

**Comment by WHUweiqingzhou** [comment]
This issue is closed after PR #3337.
